### PR TITLE
feat: unit format for insight table

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

In #11078 we added ability to set the aggregation axis to numeric, percentage, or duration for a subset of trend insights.

This extends that for Insight Tables

## Changes

* extracts a utility function so callers don't need to know if they are formatting a property, applying a chosen format, or not
* applies it to insight tables

## How did you test this code?

running it locally and seeing it happen

### CODE REVIEW

**Comment by pauldambra** [comment]
@benjackwhite @Twixes I'm going to be AFK but feel free to merge this if you're tempted to approve

**Review by clarkus — CHANGES_REQUESTED** [review]
Axis doesn't seem applicable in this case, right? Maybe just "numeric units" would be enough? I guess it depends on what all values are shown in the table and if they're all of the same format. Thoughts?

**Comment by clarkus** [comment]
I wonder if the table scenario is more closely aligned with the column configuration features we have in other places. The formatting might be specific to a column, and a user might want to format more than one column. Just a quick thought, so I could be off target here.

**Comment by pauldambra** [comment]
> Axis doesn't seem applicable in this case, right? Maybe just "numeric units" would be enough?

Ah, yes. This is out-of-date now. Imagine it says "Unit" where it currently says "aggregation axis unit" :)

Here the table is showing the series that the formatting applies to. So I think the real potential here is (at some point in the future) to allow setting a unit _per_ series

e.g. duration of event X with count of page views

At the moment we say "this is all numbers". So, now we can say "this is all numbers, or durations, or percentages"

**Review by clarkus — APPROVED** [review]

File: frontend/src/scenes/insights/views/InsightsTable/InsightsTable.tsx [hunk 1]
```diff
@@ -278,17 +282,23 @@ export function InsightsTable({
...
-                    : (value ?? 'Unknown').toLocaleString()
+
+                return value !== undefined
+                    ? formatAggregationValue(
+                          item.action?.math_property,
+                          value,
+                          (value) => formatAggregationAxisValue(filters.aggregation_axis_format || 'numeric', value),
+                          formatPropertyValueForDisplay
+                      )
+                    : 'Unknown'
```
Comment (benjackwhite) [inline frontend/src/scenes/insights/views/InsightsTable/InsightsTable.tsx:301]: I find this and the underlying function quite tough to follow... Not really a helpful comment as I don't have a great suggestion but I thought it worth saying anyways 🙈

**Review by benjackwhite — APPROVED** [review]
Approved with a comment

File: frontend/src/scenes/insights/views/InsightsTable/InsightsTable.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/views/InsightsTable/InsightsTable.tsx:301]: Yeah, that's fair... it's a yucky "do this or do that with some possibly specified default to fall back on"

The same (or at least very similar) idea is in multiple places in Insights so at the minute I've just been pushing it into one place.

Feels like maybe the problem is we pass around sets of numbers and then apply behaviour to them and we're seeing the start of there being a need to pass around sets of "things that have behaviour" 🤔

**Pull request merged**